src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -203,66 +222,203 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
     }
 
     if (!strcmp(modules_list[WM_TASK_UPGRADE_MODULE], module)) {
+        response = wm_task_manager_analyze_task_upgrade_module(task_object, error_code);
+        
+    } else if (!strcmp(modules_list[WM_TASK_API_MODULE], module)) {
+        response = wm_task_manager_analyze_task_api_module(task_object, error_code);
+        
+    } else {
+        *error_code = WM_TASK_INVALID_MODULE;
+        response = wm_task_manager_build_response(WM_TASK_INVALID_MODULE, agent_id, task_id, status);
+    }
+
+    if (status){
+        os_free(status);
+    }
+    return response;
+}
+
+cJSON* wm_task_manager_analyze_task_upgrade_module(const cJSON *task_object, int *error_code) {
+    cJSON *tmp = NULL;
+    cJSON *response = NULL;
+    int result = 0;
+    int agent_id = OS_INVALID;
+    int task_id = OS_INVALID;
+    char *status = NULL;
+    
+    char *command = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_COMMAND])->valuestring;
+    char *module = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_MODULE])->valuestring;
+
+    if (tmp = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_AGENT_ID]), tmp && tmp->type == cJSON_Number) {
```

Comment:
This code is repeated. You should add `agent_id`, `task_id` and `status` as parameters of this function.

Reply:
you're right